cmake/Components.cmake

```diff
@@ -0,0 +1,146 @@
+include(CMakeParseArguments)
+
+set(global_prefix "${PROJECT_NAME}_")
+set(component_prefix "${PROJECT_NAME}_component_")
+
+define_property(GLOBAL PROPERTY "${global_prefix}INCLUDE_DIRS"
+  BRIEF_DOCS "Global include directories used by all components."
+  FULL_DOCS "Global include directories used by all components."
+)
+define_property(GLOBAL PROPERTY "${global_prefix}COMPONENTS"
+  BRIEF_DOCS "List all known Aikido components."
+  FULL_DOCS "List all known Aikido components."
+)
+
+#==============================================================================
+function(add_component component)
```

Comment:
> Can we turn this into a metafunction using CMakeParseArguments so that we have single declarations that look like: <snip>

I am open to adding syntactic sugar around `add_component`, but I would prefer to do that in a future pull request.

> Also, is there a need to make this `add_aikido_component()`?

I am not sure what to do here. I intentionally _did not_ add an `aikido` prefix to these variables because I would like to reuse the same `Components.cmake` file in other projects (e.g. DART). However, this does create the potential for name collisions.

Is there any way to namespace a `.cmake` file? I attempted to define the functions like this:
```cmake
function(${prefix}_add_component)
  # ...
endfunction()
```
Unfortunately, I couldn't figure out how to _call_ that function from this file. These did *not* work:
```cmake
${prefix}_add_component() # nope
"${prefix}_add_component"() # nope
```